src/back/spv/writer.rs

```diff
@@ -811,13 +827,49 @@ impl Writer {
                         .unwrap();
                     constituent_ids.push(component_id);
                 }
+                let constituent_ids_slice = constituent_ids.as_slice();
+
+                let id = match &ir_module.types[*ty].inner {
```

Comment:
let's not match on a reference please!
Instead, add `ref` near the non-movable fields. It's clear to the reader what's going on therefore

Reply:
Always forgetting that, I think you already told me 10 times! Really need to think about that, change incoming in 1 minute.